Certificates/cert-gen.py

```diff
@@ -0,0 +1,103 @@
+# Requires inkscape in path and PDF module
+import argparse
+import csv
+import numpy as np
+import os
+import pathlib
+from PDF import PdfFileReader, PdfFileWriter
+from subprocess import run
+
+def argument_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'template_svg',
+        help="Input svg template with text matching 'NAME OF STUDENT IN CAPS' and 'NAME OF SCHOOL IN CAPS' for "
+             "replacement with the students' names and schools'/teams' names, respectively, "
+             "on the produced certificates.",
+        type=pathlib.Path,
+    )
+    parser.add_argument(
+        '-i', '--input_data',
+        help="Input csv with columns of 'First Name', 'Last Name', and 'School'."
+             "In absence of this argument, the script's own test data will be used.",
+        nargs=1,
+        type=pathlib.Path,
+    )
+    parser.add_argument(
+        '-o','--output_path',
+        help="Path in which the output will be deposited. Defaults to current working directory.",
+        type=pathlib.Path,
+        nargs=1,
+        default=os.getcwd(),
+    )
+    return parser
+
+def main():
+    options = argument_parser().parse_args()
+    
+    outpath = pathlib.Path(options.output_path, "Certificates.pdf")
+    tempsvgpath = pathlib.Path(options.output_path, ".temp-cert.svg")
+    temppdfpath = pathlib.Path(options.output_path, ".temp-cert.pdf")
```

Comment:
Do these really need to be temporary _files_? If so, do they need to actually be in the output path?

In general I would encourage avoiding interacting with any files unless necessary. This is partly because doing so is slow (admittedly not noticeable here, but in the general case writing a file is several hundred times slower than going to memory for the same data), but also because it means that the code needs to worry about where to put the files and ensuring that they are cleaned up properly.

If these really do need to be temporary _files_ (because, for example, we're passing them to another process), I would encourage use of the [`tempfile`](https://docs.python.org/3/library/tempfile.html) module to manage them. That lets you make use of the context manager (`with` statement) to ensure that they're removed when you're done with them.

However, if all you need is a file-like thing, you might be able to use an [`io.StringIO`](https://docs.python.org/3/library/io.html#io.StringIO) or an [`io.BytesIO`](https://docs.python.org/3/library/io.html#io.BytesIO). These keep the actual data in memory, but otherwise look like files from Python.

Reply:
So far as I can see, it does require actual files, but tempfile seems pretty awesome :)